esp-hal/src/timer/systimer.rs

```diff
@@ -146,90 +93,81 @@ impl<'d> SystemTimer<'d> {
     }
 
     /// Create a new instance.
-    pub fn new(_systimer: impl Peripheral<P = SYSTIMER> + 'd) -> Self {
+    pub fn new(_systimer: SYSTIMER) -> Self {
         // Don't reset Systimer as it will break `time::now`, only enable it
         PeripheralClockControl::enable(PeripheralEnable::Systimer);
 
         #[cfg(soc_etm)]
         etm::enable_etm();
 
         Self {
-            unit0: SpecificUnit::new(),
-            #[cfg(not(esp32s2))]
-            unit1: SpecificUnit::new(),
-            comparator0: SpecificComparator::new(),
-            comparator1: SpecificComparator::new(),
-            comparator2: SpecificComparator::new(),
+            alarm0: Alarm::new(0),
+            alarm1: Alarm::new(1),
+            alarm2: Alarm::new(2),
         }
     }
 
-    /// Get the current count of Unit 0 in the System Timer.
-    pub fn now() -> u64 {
+    /// Get the current count of the given unit in the System Timer.
+    pub fn unit_count(unit: Unit) -> u64 {
```

Comment:
Counter, please, otherwise I'd expect this to return 2 :D

Reply:
I went with `unit_value` instead, counter didn't feel correct either.